lib/services/files-api-service.js

```diff
@@ -0,0 +1,129 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+var fs = require('fs');
+var path = require('path');
+var walking = require('walk');
+
+module.exports = filesApiServiceFactory;
+
+di.annotate(filesApiServiceFactory, new di.Provide('Http.Services.Api.StaticFiles'));
+di.annotate(filesApiServiceFactory,
+    new di.Inject(
+      'Constants',
+      'Promise',
+      'Services.Configuration',
+      'Services.Waterline'
+    )
+);
+
+
+function filesApiServiceFactory(
+    constants,
+    Promise,
+    configuration,
+    waterline
+) {
+    var walker;
+    var walk = Promise.promisifyAll(walking);
+    var skuPath = constants.HttpStatic.onhttpSkupack;
+    var staticPath = constants.HttpStatic.onhttpStaticHttp;
+    Promise.promisifyAll(fs);
+
+    function FilesApiService() {
+    }
+
+
+    FilesApiService.prototype.pairSkupackIds = function() {
+        return fs.readdirAsync(skuPath).then(function(list) {
```

Comment:
I've got it down to this. Sparing a push in progress, is there more reduction that I'm missing or care for things like which line the `then` statements are on?
```
StaticFilesApiService.prototype.pairSkupackIds = function() {
    return fs.readdirAsync(skuPath).filter(function(item) {
        return (fs.lstatAsync(path.join(skuPath, item)))
        .then(function(stat) {
            return stat.isDirectory();
        }); 
    }).reduce(function(result, item) {
        return waterline.skus.findOne({id : item})
        .then(function(sku) {
            result[item] = sku.name;
            return result;
        }); 
    },{});
};
```